fsspec/utils.py

```diff
@@ -500,7 +554,7 @@ def merge_offset_ranges(paths, starts, ends, max_gap=0, max_block=None, sort=Tru
     if not isinstance(starts, list):
         starts = [starts] * len(paths)
     if not isinstance(ends, list):
-        ends = [starts] * len(paths)
+        ends = [ends] * len(paths)
```

Comment:
I assume this was just a bug? I can't figure out what this could would do otherwise.

Reply:
@rjzamora  ?

Reply:
Certainly looks like a bug to me. That said, I don't think `starts` or `ends` ever correspond to anything other than `list` anyway. Perhaps we can just drop the `int` support since the result would have been "wrong" before anyway?